pkg/playbook/app/domain/connection.go

```diff
@@ -104,30 +104,63 @@ func isValidTpp(c Connection) (bool, error) {
 }
 
 func isValidVaaS(c Connection) (bool, error) {
-	// Credentials are not empty
+	// Check if an API key has been provided
 	apikey := false
 	if c.Credentials.APIKey != "" {
 		apikey = true
 	}
 
-	svcaccount := false
+	accesstoken := false
+	if c.Credentials.AccessToken != "" {
+		accesstoken = true
+	}
+
+	// Check if an TokenURL has been provided
+	tokenurl := false
 	if c.Credentials.TokenURL != "" {
+		tokenurl = true
+	}
+
+	// Check if externalJWT has been provided
+	externalJWT := false
+	if c.Credentials.ExternalJWT != "" {
+		externalJWT = true
+	}
+
+	// There's a valid service account IF both externalJWT and tokenURL provided
+	svcaccount := false
+	if externalJWT && tokenurl {
 		svcaccount = true
+	} else if externalJWT && !tokenurl {
+		// JWT Provided without token url
+		return false, ErrNoVCPTokenURL
+	} else if tokenurl && !externalJWT {
+		// Tokenurl without an external JWT
+		return false, ErrNoExternalJWT
 	}
 
-	if !apikey && !svcaccount {
-		return false, ErrNoCredentials
+	// If there's just an API key provided, that's good
```

Comment:
Your logic here is a bit too complex.
I think what you're trying for here (after line 142) is:
```
if !svcaccount && !apikey && !accesstoken {
	// none of the credential options were provided
	return false, ErrNoCredentials
}
// if we got here then at least one of the credential options was provided
if (svcaccount && apikey) || (svcaccount && accesstoken) || (apikey && accesstoken) {
	// more than one credential option is not acceptable
	return false, ErrAmbiguousVCPCreds
}
// if we got here then only one credential option was provided (which is what we want)
return true, nil
```

Reply:
much better, thanks